### ISSUE TITLE
Top navigation tabs [MVP]

### ISSUE THREAD

**Issue by jarekmorawski** [woothemes/woocommerce#35706]
In preparation for introducing variation management to the new product editing experience, we want to add tabs to the top bar. We'll start small and only accommodate what is needed for the MVP, but the acceptance criteria mention future improvements since they can impact how we structure code from the very start.

**Tab content (only MVP scope)**

- General
   - Product details
   - Images
   - Attributes
- Pricing
   - Pricing
- Inventory
   - Inventory
- Shipping
   - Shipping
- Options
   - Options
   - Variations

**Design**

`5sAIeTRd9Yp7nSCT33BAWz-fi-5311%3A197937&t=4r3ndH9ltPGnsjGr-4`

<img width="601" alt="image" src="https://user-images.githubusercontent.com/79307566/203767462-e13055e2-a0df-4348-a172-21f0c997849e.png">

**Acceptance criteria**

- [ ] Users can see tabs at the top of the screen. They're part of the top navigation bar.
- [ ] The tabs are fixed at the top of the screen when users scroll up and down the page.
- [ ] On mobile, the tabs scroll horizontally. Design: `5sAIeTRd9Yp7nSCT33BAWz/%5BMaster%5D-fi-7342%3A344251&t=4r3ndH9ltPGnsjGr-4`
- [ ] The page doesn't reload when users switch between tabs.
- [ ] We don't remember the scroll position when users switch between tabs. They always start at the top.
- [ ] Each tab has multiple states: resting (default), selected (with the `Gutenberg-Blue` underline), and disabled (`Gutenberg-600`, `cursor: not-allowed;`)
- [ ] Each tab can have a suffix element, such as a badge or notification dot (not part of the MVP).
- [ ] Each tab is an identifiable element, and we can target it with the spotlight component.
- [ ] The tab bar can be expanded with custom tabs added by third-party plugins.

**Comment by joshuatf** [comment]
This issue is blocked by https://github.com/woocommerce/woocommerce/issues/35766.  To avoid further duplication in tab logic, we should make sure that https://github.com/woocommerce/woocommerce/issues/35766 is completed first.

The [TabPanel](https://developer.wordpress.org/block-editor/reference-guides/components/tab-panel/) is a good candidate for this feature and includes most of the logic we need already implemented.

However, there are 2 challenges with this approach:

1. In order to fix the tabs underneath the header, we'll have to use some CSS magic.  This should respect the margin added to `#wpbody` if possible as this represents the WC header height which should continue to work if the header height was changed.
2. The way the `tabs` props works in the `TabPanel` component is not easily extensible with SlotFill.  This is something we should follow-up on in the future during our extensibility spike.

**Comment by joshuatf** [comment]
Estimate: 2
